Legacy review of RabbitMQ message id strategy (#4132)
Particular/docs.particular.net: transports/rabbitmq/message-id-strategy.md

---

By default, the `message-id` property of the AMQP standard is used to relay the [message identity](/nservicebus/messaging/message-identity.md). If this property isn't set, the transport will throw an exception because NServiceBus requires a message identity to perform retries, de-duplication, etc., in a safe way. 

For integration scenarios where the sender is not controlled, the receiver might need to employ a custom strategy that extracts a message identity from a custom header or part of the message body. This custom strategy can be configured by calling:

snippet: rabbitmq-config-custom-id-strategy

> WARNING: It is extremely important to ensure that the custom strategy is deterministic i.e. that it returns the same value when invoked multiple times for the same message, and that no two messages are associated the same value. 

Reviewer: ```suggestion
WARNING: It is extremely important that the custom strategy is deterministic (it returns the same value when invoked multiple times for the same message), and that no two messages are assigned the same value. 
```
